src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/SourceControlTests.ps1

```diff
@@ -0,0 +1,322 @@
+{
+    # VsoGit repo info
+    $sourceControl = @{
+        Name = "AASourceControl-VsoGit"
+        RepoUrl = "https://francisco-gamino.visualstudio.com/_git/VsoGit-SwaggerAndCmdletsTests"
+        Branch = "preview"
+        FolderPath = "Azure/MyRunbooks"
+        SourceType = "VsoGit"
+    }
+
+    try
+    {
+        EnsureSourceControlDoesNotExist -Name $sourceControl.Name
+
+        # Access token
+        $token = "3qdxa22lutnhezd4atpna74jn3m7wgo6o6kfbwezjfnvgbjhvoca"
+        $accessToken = ConvertTo-SecureString -String $token -AsPlainText -Force
+
+        $createdSourceControl = New-AzureRmAutomationSourceControl -ResourceGroupName $resourceGroupName `
+                                                                   -AutomationAccountName $automationAccountName  `
+                                                                   -Name $sourceControl.Name  `
+                                                                   -RepoUrl $sourceControl.RepoUrl `
+                                                                   -Branch $sourceControl.Branch `
+                                                                   -SourceType $sourceControl.SourceType `
+                                                                   -FolderPath $sourceControl.FolderPath `
+                                                                   -AccessToken $accessToken `
+                                                                   -DoNotPublishRunbook
+
+        # Validate that the source control was created
+        Assert-NotNull $createdSourceControl "Failed to create VsoGit source control."
+
+        # Validate the source control properties
+        $propertiesToValidate = @("Name", "RepoUrl", "SourceType", "Branch", "FolderPath")
+        
+        foreach ($property in $propertiesToValidate)
+        {
+            Assert-AreEqual $sourceControl.$property $createdSourceControl.$property `
+                "'$property' of created source control does not match. Expected:$($sourceControl.$property). Actual: $($createdSourceControl.$property)"
+        }
+
+        # Enable PublishRunbook in the newly created source control 
+        $updatedSourceControl = Update-AzureRmAutomationSourceControl -ResourceGroupName $resourceGroupName `
+                                                                      -AutomationAccountName $automationAccountName  `
+                                                                      -Name $sourceControl.Name  `
+                                                                      -PublishRunbook $true
+        $expectedPropertyValue = "True"
+        Assert-AreEqual $updatedSourceControl.PublishRunbook $expectedPropertyValue `
+                "'PublishRunbook' property does not match. Expected: $expectedPropertyValue. Actual: $($updatedSourceControl.PublishRunbook)"
+        
+        # Start a sync for the source control
+        $jobId = "0bfa6b49-c08c-4b2f-853e-08128c3c86ee"
```

Comment:
This should not be hardcoded - you should be able to query for the job id

Reply:
This job id is part of the Url for the call. Originally, I was creating one on the fly (New-Guid).Guid, but this was causing my reply to fail.